cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -1084,22 +1084,63 @@ bool TrainBatchTaskBase<algorithmFPType, BinIndexType, DataHelper, cpu>::findBes
                                                                                              typename DataHelper::TSplitData & bestSplit,
                                                                                              algorithmFPType totalWeights)
 {
-    chooseFeatures();
-    size_t nVisitedFeature       = 0;
-    const size_t maxFeatures     = nFeatures();
-    const float qMax             = 0.02; //min fracture of observations to be handled as indexed feature values
-    IndexType * bestSplitIdx     = featureIndexBuf(0) + iStart;
-    IndexType * aIdx             = _aSample.get() + iStart;
-    int iBestSplit               = -1;
-    int idxFeatureValueBestSplit = -1; //when sorted feature is used
+    /* counter of the number of visited features, we visit _nFeaturesPerNode
+    *  depending on _par.useConstFeatures constant features can be skipped
+    */
+    size_t nVisitedFeature = 0;
+    /* total number of features */
+    const size_t maxFeatures = nFeatures();
+    /* (??) min fracture of observations to be handled as indexed feature values */
+    const float qMax = 0.02;
+    /* index of the best split, initialized to first index we investigate */
+    IndexType * bestSplitIdx = featureIndexBuf(0) + iStart;
+    /* sample index */
+    IndexType * aIdx = _aSample.get() + iStart;
+    /* zero-based index of best split */
+    int iBestSplit = -1;
```

Comment:
Using integers and any size specified types (i.e. `int` instead of `std::int32_t`) is generally banned from library. At least some sanity checks and analysis should be performed to guarantee that integer overflow doesn't take place here.

Reply:
As you probably noticed, this also has been around for a long time. If we use `BinIndexType` instead, we're save. See `df_classification_train_dens_default_impl.i`, where `indexedFeatures.maxNumIndices()` is checked _or_ `dtrees::internal::IndexedFeatures::IndexType` is used.

Not, however, that `IndexType` has a todo -- not sure if that's relevant here

```c++
// found in dtrees_feature_type_helper.h:89
    typedef int IndexType; // TODO: should be unsigned int
```

Reply:
I don't know either. Maybe it should make this code supporting larger number of samples.

Reply:
Should I change it?